adafruit_mcp2515/__init__.py

```diff
@@ -217,6 +246,7 @@ def __init__(
         cs_pin,
         *,
         baudrate: int = 250000,
+        xtal_frequency: int = 16000000,
```

Comment:
I would change the typing to use `Literal` to denote that only specific values are expected.  I would also also consider changing the name to `crystal_freq` since I think the abbreviation makes more sense on the other word, but that's more of a personal suggestion.

```suggestion
        crystal_freq: Literal[8000000, 16000000] = 16000000,
```

You'll need to import `typing_extenstions` in a `try`/`except` block [like we do in other libraries](https://github.com/adafruit/Adafruit_CircuitPython_ADS1x15/blob/1554279746863bf07b7e9f770cd9f62c33d027df/adafruit_ads1x15/ads1015.py#L15-L20).  You'll need the `try`/`except` because `typing_extensions` isn't available on CircuitPython, but type annotations are not parsed during runtime so it's fine.

Reply:
Solved on last PR.